IdentityCore/src/validation/MSIDB2CAuthority.m

```diff
@@ -165,4 +165,12 @@ + (NSURL *)normalizedAuthorityUrl:(NSURL *)url
     return [NSURL URLWithString:normalizedAuthorityUrl];
 }
 
+#pragma mark - NSCopying
+
+- (id)copyWithZone:(NSZone *)zone
+{
+    MSIDB2CAuthority *authority = [[self.class allocWithZone:zone] initWithURL:_url validateFormat:NO context:nil error:nil];
```

Comment:
shouldn't we also copy openid connect endpoint and metadata?

Reply:
yeeeeeeeesssssss